src/response/file.rs

```diff
@@ -231,19 +231,16 @@ impl Stream for FileStream {
 
             Poll::Ready(Ok(())) => {
                 let filled = data.filled();
+                let len = filled.len();
 
-                if let Some(remaining) = this.remaining.checked_sub(filled.len()) {
+                if let Some(remaining) = this.remaining.checked_sub(len) {
                     let data = Bytes::copy_from_slice(filled);
                     this.remaining = remaining;
-
-                    if remaining == 0 {
-                        this.buffer.fill(MaybeUninit::uninit());
-                    }
-
+                    this.buffer[..len].fill_with(MaybeUninit::uninit);
```

Comment:
I recently changed my mind about preferring `(&mut this.bufer[..len])` after thinking about it and realizing this happens anytime that a method that takes `&self` is called from method that takes an owned version of `self`.